CMakeLists.txt

```diff
@@ -104,7 +104,14 @@ list(APPEND CMAKE_MODULE_PATH "${TRITON_ROOT}/CMakeModules/")
 if(Z3_INTERFACE)
     message(STATUS "Compiling with Z3 SMT solver")
     find_package(Z3 REQUIRED)
-    include_directories(${Z3_INCLUDE_DIRS})
+    message(STATUS "Z3 version: ${Z3_VERSION}")
+    if(TARGET z3::libz3)
+        link_libraries(z3::libz3)
+    elseif(DEFINED Z3_INCLUDE_DIRS)
+        include_directories(${Z3_INCLUDE_DIRS})
```

Comment:
I do not understand why it's an `elseif` there? `link_libraries` and `include_directories` have not the same behavior?

Reply:
Not in this case `z3::libz3` is a CMake target describing _how_ to link to it, meaning it describes what are the include directories to include and which libraries to link to.
The way to leverages those description from a CMake target is by using `(target_)link_libraries` in CMake.

Reply:
`include_directories` CMake command expected path which will simply be added to the compiler command line accordingly, there is no "smart resolution" like when using `(target_)link_librarie` with CMake targets